File: TheepotMobile/TheepotMobile/Models/Kilometerstanden.swift
import Foundation

enum VoertuigType: String, Codable {
    case auto, bus
}

enum RegelmaatEenheid: String, Codable {
    case week, maand, kwartaal
}

/// Tabel `km_voertuigen`.
struct KmVoertuig: Codable, Identifiable, Hashable {
    let id: String
    let kenteken: String
    let type: VoertuigType
    let omschrijving: String?
    let actief: Bool
    let regelmaatAantal: Int
    let regelmaatEenheid: RegelmaatEenheid

    enum CodingKeys: String, CodingKey {
        case id, kenteken, type, omschrijving, actief
        case regelmaatAantal = "regelmaat_aantal"
        case regelmaatEenheid = "regelmaat_eenheid"
    }
}

/// Tabel `km_registraties`.
struct KmRegistratie: Codable, Identifiable {
    let id: String
    let voertuigId: String
    var kilometerstand: Int
    var datum: String
    var notitie: String?
    let ingevoerdDoor: String?

    enum CodingKeys: String, CodingKey {
        case id
        case voertuigId = "voertuig_id"
        case kilometerstand, datum, notitie
        case ingevoerdDoor = "ingevoerd_door"
    }
}

File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/models/Kilometerstanden.kt
package nl.bsodetheepot.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class VoertuigType {
    @SerialName("auto") AUTO,
    @SerialName("bus") BUS,
}

@Serializable
enum class RegelmaatEenheid {
    @SerialName("week") WEEK,
    @SerialName("maand") MAAND,
    @SerialName("kwartaal") KWARTAAL,
}

/** Tabel `km_voertuigen`. */
@Serializable
data class KmVoertuig(
    val id: String,
    val kenteken: String,
    val type: VoertuigType,
    val omschrijving: String? = null,
    val actief: Boolean = true,
    @SerialName("regelmaat_aantal") val regelmaatAantal: Int = 1,
    @SerialName("regelmaat_eenheid") val regelmaatEenheid: RegelmaatEenheid = RegelmaatEenheid.MAAND,
) {
    val label: String get() = "$kenteken (${if (type == VoertuigType.BUS) "Bus" else "Auto"})"
}

/** Tabel `km_registraties`. */
@Serializable
data class KmRegistratie(
    val id: String,
    @SerialName("voertuig_id") val voertuigId: String,
    val kilometerstand: Int,
    val datum: String,
    val notitie: String? = null,
    @SerialName("ingevoerd_door") val ingevoerdDoor: String? = null,
)
